Remove Docker references in image concept page (#31895)
kubernetes/website: content/en/docs/concepts/containers/images.md

If you don't specify a registry hostname, Kubernetes assumes that you mean the Docker public registry.

> After the image name part you can add a _tag_.

Reviewer: @a-mccarthy I think the original text is OK. After dockershim is removed, people can still use commands like `docker` and `podman` locally with containers, and those analogies will remain relevant.

Author: @sftim i was going off the info in the issue, which called out removing references to podman, but i can add them back. I also removed the wording bc i felt like it was more confusing. But if the examples are still valid, i can add them back in

> After the image name part you can add a _tag_ (in the same way you would when using with commands like `docker` or `podman`).

Reviewer 2: nit: Tidy up the sentence?
A container image name may include a `tag` component. To add a `tag` to an image, you can use `docker` or `podman`.

### Configuring nodes to authenticate to a private registry

> Specific instructions for setting credentials depends on the container runtime and registry you chose to use. You should refer your solution's documentation for the most accurate information.

Reviewer: ```suggestion
Specific instructions for setting credentials depends on the container runtime and registry you chose to use. You should refer to your solution's documentation for the most accurate information.
```
?

> For an example configuring a private Docker registry, see the [Pull an Image from a Private Registry](/docs/tasks/configure-pod-container/pull-image-private-registry) task.

Reviewer: ```suggestion
For an example of configuring a private container image registry, see the
[Pull an Image from a Private Registry](/docs/tasks/configure-pod-container/pull-image-private-registry)
task. That example uses a private registry in Docker Hub.
```
?

> #### Creating a Secret with a Docker config

Author: Do we also want to keep this example here? or point to the pulling images from a registry page as well? https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials

Reviewer: Hmm. https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials doesn't exactly match https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets (different Secret types).

I suggest we omit that fix from _this_ PR and open a new, separate issue about harmonising the docs for this task. The “Docker” element here is more about either having `docker` on your PR or using Docker Hub to store images&mdash;the changes would be applicable to any container runtime that can pull images from a docker-style image repository.
That new issue should be prioritised important-soon, IMO.

Author: Ok, @sftim i can leave in this example. I can open a new issues, but I'm not sure i understand fully what the recommended action should be for the new issue you are proposing. Can you explain a little more?

Reviewer: If you tag me on that new issue @a-mccarthy then I will try to check it and make sure it matches what I'd hoped to explain.

#### Creating a Secret with a Docker config

> Using a Docker registry as an example, run the following command, substituting the appropriate uppercase values:

Reviewer: ```suggestion
You need to know the username, registry password and client email address for authenticating
to the registry, as well as its hostname.
Run the following command, substituting the appropriate uppercase values:
```

The “Docker” bit here is that the contents of the Secret are compatible with Docker. The container registry doesn't have to be Docker Hub.
